compositor_pipeline/src/audio_mixer/internal_audio_mixer.rs

```diff
@@ -101,6 +102,22 @@ impl InternalAudioMixer {
             .collect()
     }
 
+    // Produce continuous batch of samples for range (start_pts, end_pts).
```

Comment:
Yep, and does like 10 different input validations along the way. I don't like those validations in the middle of this merging logic. I understand why you want to validate if the queue output is right (if I wrote it, I wouldn't be sure either :P), but this should be done before calling this function, not in the middle of it